CMakeLists.txt

```diff
@@ -181,25 +262,66 @@ add_library(
     NEMS/src/nems_c_rusage.c
     NEMS/src/ENS_Cpl/ENS_CplComp_ESMFMod_STUB.F90
 )
-add_dependencies(ufs-s2s-model-lib fv3cap)
-target_link_libraries(ufs-s2s-model-lib PRIVATE esmf)
+add_dependencies(ufs-weather-model-lib fv3atm fms)
+target_link_libraries(ufs-weather-model-lib PRIVATE esmf)
+
+target_compile_definitions(ufs-weather-model-lib PRIVATE -DESMF_VERSION_MAJOR=${ESMF_VERSION_MAJOR})
+target_compile_definitions(ufs-weather-model-lib PRIVATE -DFRONT_FMS)
+target_compile_definitions(ufs-weather-model-lib PRIVATE -DFRONT_FV3=fv3gfs_cap_mod)
 
-target_compile_definitions(ufs-s2s-model-lib PRIVATE -DESMF_VERSION_MAJOR=${ESMF_VERSION_MAJOR})
-target_compile_definitions(ufs-s2s-model-lib PRIVATE -DFRONT_FMS)
-target_compile_definitions(ufs-s2s-model-lib PRIVATE -DFRONT_FV3=fv3gfs_cap_mod)
+target_include_directories(ufs-weather-model-lib PRIVATE ${PROJECT_BINARY_DIR}/NEMS/src)
+target_include_directories(ufs-weather-model-lib PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/FV3/mod)
+target_include_directories(ufs-weather-model-lib PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/FMS/mod)
 
-target_include_directories(ufs-s2s-model-lib PRIVATE ${PROJECT_BINARY_DIR}/NEMS/src)
-target_include_directories(ufs-s2s-model-lib PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/FV3/mod)
+if(CCPP)
+    target_compile_definitions(ufs-weather-model-lib PRIVATE -DFRONT_CCPP)
+endif()
 
 if(WW3)
-    target_compile_definitions(ufs-s2s-model-lib PRIVATE -DFRONT_WW3=WMESMFMD)
-    target_include_directories(ufs-s2s-model-lib PRIVATE ${PROJECT_SOURCE_DIR}/WW3/model/mod)
-    set_target_properties(ufs-s2s-model-lib PROPERTIES Fortran_MODULE_DIRECTORY ${PROJECT_SOURCE_DIR}/WW3/model/mod)
-    add_dependencies(ufs-s2s-model-lib ww3_nems)
+    target_compile_definitions(ufs-weather-model-lib PRIVATE -DFRONT_WW3=WMESMFMD)
+    target_include_directories(ufs-weather-model-lib PRIVATE ${PROJECT_SOURCE_DIR}/WW3/model/mod)
+    set_target_properties(ufs-weather-model-lib PROPERTIES Fortran_MODULE_DIRECTORY ${PROJECT_SOURCE_DIR}/WW3/model/mod)
+    add_dependencies(ufs-weather-model-lib ww3_nems)
+    target_link_libraries(ufs-weather-model-lib PUBLIC ${WW3_LIBS})
+    target_include_directories(ufs-weather-model-lib PUBLIC
+                                       $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/WW3/model/mod>)
+endif()
+
+if(MOM6)
+    target_compile_definitions(ufs-weather-model-lib PRIVATE -DFRONT_MOM6=mom_cap_mod)
+    target_include_directories(ufs-weather-model-lib PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/MOM6-interface/mod)
+    set_target_properties(ufs-weather-model-lib PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/MOM6-interface/mod)
+    add_dependencies(ufs-weather-model-lib nuopcmom6)
+    target_link_libraries(ufs-weather-model-lib PUBLIC nuopcmom6)
+    target_include_directories(ufs-weather-model-lib PUBLIC
+                                       $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/MOM6-interface/mod>)
+    list(APPEND optional_targets nuopcmom6)
```

Comment:
this is not optional in S2S

Reply:
I'll edit so that if(S2S) sets these as s2s_targets